src/components/Field.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import styles from "./Field.module.css";
+
+const PASSWORD_ICON_CONFIG = {
+  false: {
+    src: "./images/icon_password_invisible.png",
+    className: "password-icon hidden",
+  },
+  true: {
+    src: "./images/icon_password_visible.png",
+    className: "password-icon",
+  },
+};
+
+const INPUT_CONTAINER_CLASSNAME = {
+  null: "",
+  false: "invalid",
+  true: "valid",
+};
+
+const Field = ({
+  fieldConfig,
+  value,
+  valid,
+  hint,
+  handleInputChange,
+  handleInputBlur,
+}) => {
+  //prettier-ignore
+  const [isVisible, setIsVisible] = useState(false);
+  const handlePasswordIconClick = () => setIsVisible((prev) => !prev);
```

Comment:
단순히 토글용이라면 아래와 같은 형태가 더 간단합니다.

```suggestion
  const handlePasswordIconClick = () => setIsVisible(!isVisible);
```